File: python/Leetcode278.py
class Solution:
    def firstBadVersion(self, n: int) -> int:
        # Just added for removing red lines in the editor
        def isBadVersion(n):
            return True

        fi = 1
        while fi < n:
            mid = (fi + n) // 2
            if isBadVersion(mid):
                n = mid - 1
            else:
                fi = mid + 1
        if isBadVersion(n):
            return n
        else:
            return n + 1

File: java/src/Leetcode278.java
public class Leetcode278 {
    public int firstBadVersion(int n){
        int first = 1;
        
        while(first < n){
            int mid = first + (n - first) / 2;
            if(isBadVersion(mid)){
                n = mid - 1;
            }else
                first = mid + 1;
        }

        if(isBadVersion(n))
            return n;
        else
            return n + 1;
    }

    private boolean isBadVersion(int n){
        return true;
    }
}
